### PR TITLE
migrate kwok jobs to community clusters

### PR DESCRIPTION
**Pull request by kranurag7**
migrate kubernetes-sigs/kwok jobs to community clusters 

Ref: #29722

### CODE REVIEW

**Comment by rjsadow** [comment]
Thanks for submitting this @kranurag7! There is one modification that'll be needed for each of the jobs before we're ok to test. The Google-owned clusters did not require any resource definitions whereas the community-owned clusters do. For each of the containers in this job, can you please work with the kwok owners (@Huang-Wei and @wzshiming) to determine what good memory/cpu quotas would be?

File: config/jobs/kubernetes-sigs/kwok/kwok-presubmits-main.yaml [hunk 1]
```diff
@@ -55,6 +58,7 @@ presubmits:
   - name: pull-kwok-e2e-test-main
     decorate: true
     path_alias: sigs.k8s.io/kwok
+    cluster: eks-prow-build-cluster
```
Comment (wzshiming) [inline config/jobs/kubernetes-sigs/kwok/kwok-presubmits-main.yaml:61]:
``` yaml
          requests:
            memory: "1Gi"
            cpu: "1"
          limits:
            memory: "2Gi"
            cpu: "2"
```

I'm not sure of the exact request but I think it should be enough, if not I'll add more 😄.

Comment (rjsadow) [reply config/jobs/kubernetes-sigs/kwok/kwok-presubmits-main.yaml:61]: One other point of note, the `pull-test-infra-unit` requires guaranteed QoS so the requests will need to match the limits.

Comment (wzshiming) [reply config/jobs/kubernetes-sigs/kwok/kwok-presubmits-main.yaml:61]: Oh thanks

``` yaml
          requests:
            memory: "2Gi"
            cpu: "2"
          limits:
            memory: "2Gi"
            cpu: "2"
```

**Comment by rjsadow** [comment]
@kranurag7 are you still working on this PR?

**Comment by kranurag7** [comment]
@rjsadow Thanks for the ping, I'll update the PR today.

File: config/jobs/kubernetes-sigs/kwok/kwok-presubmits-main.yaml (hunk 1 above)
Comment (kranurag7) [reply config/jobs/kubernetes-sigs/kwok/kwok-presubmits-main.yaml:82]: @rjsadow After specifying the `resources` field, the [tests](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29731/pull-test-infra-unit-test/1668709627019137024#1:build-log.txt%3A553-576) are not able to detect the  changes.

**Pull request merged**